### ISSUE TITLE
[QST] Why always install package from GitHub when running cuvs/examples/build.sh?

### ISSUE THREAD

**Issue by Wang-Zm** [rapidsai/cuvs#291]
**What is your question?**

When I first execute the cuvs/examples/build.sh, some packages are installed by CPM.cmake from GitHub. When I modify the source files in cuvs/examples/cpp/src, I try to execute build.sh to compile, but the packages still need to be added by CPM, which costs a lot of time. I want to know how to avoid installing packages repeatedly.

Thanks in advance.

**Comment by cjnolet** [comment]
Hi @Wang-Zm, that is unexpected behavior. Generally, if you have RAFT installed already which has the same RAPIDS version as cuVS (by either building and installing it from source, or by installing it into a conda environment) cmake should not attempt to build RAFT at all. I believe the problem you might be encountering is that RAFT was never actually installed since it was built as a dependency and this is causing it to want to build every time. @robertmaynard is this expected behavior when RAFT hasn't been installed previously? 

Can you try installing RAFT before building cuVS and see if rebuilding cuVS uses the installed version of RAFT instead of rebuilding it with cuVS?

**Comment by robertmaynard** [comment]
Yes the expected behavior is that cuvs and dependencies will be built from source if they aren't found locally. So in this case you would need cuvs and raft installed locally and discoverable by CMake ( e.g. in your current conda env ) for them to be used.

The major thing to note is that if you are building the latest version of the examples it would mean you need cuvs 24.10 and raft 24.10 would would require using our nightly packages.
